src/adapters/morpho-compound-v2/common/morphoBasePoolAdapter.ts

```diff
@@ -0,0 +1,699 @@
+        break
+      case 'borrowed':
+        filter = morphoCompoundContract.filters.Borrowed(userAddress)
+        break
+    }
+
+    const eventResults = await morphoCompoundContract.queryFilter(
+      filter,
+      fromBlock,
+      toBlock,
+    )
+
+    const movements = await Promise.all(
+      eventResults.map(async (event) => {
+        const castedEvent = this._castEventToLogType(event, eventType)
+        const eventData = this._extractEventData(castedEvent)
+        if (!eventData) {
+          return null
+        }
+
+        const protocolToken = await this._fetchProtocolTokenMetadata(
+          eventData._poolToken,
+        )
+        const underlyingTokens = await this._fetchUnderlyingTokensMetadata(
+          eventData._poolToken,
+        )
+
+        const underlyingTokensMovement: Record<string, BaseTokenMovement> = {}
+        underlyingTokens.forEach((underlyingToken) => {
+          underlyingTokensMovement[underlyingToken.address] = {
+            ...underlyingToken,
+            transactionHash: event.transactionHash,
+            movementValueRaw: eventData._amount,
+          }
+        })
+
+        return {
+          protocolToken: {
+            ...protocolToken,
+          },
+          underlyingTokensMovement,
+          blockNumber: event.blockNumber,
+        }
+      }),
+    )
+    return movements.filter(
+      (movement): movement is MovementsByBlock => movement !== null,
+    ) as MovementsByBlock[]
+  }
+
+  protected async _getProtocolTokenApr({
+    protocolTokenAddress,
+    blockNumber,
+  }: GetAprInput): Promise<number> {
+    const lensContract = MorphoCompoundLens__factory.connect(
+      this.lensAddress,
+      this._provider,
+    )
+    const positionType = this.getProtocolDetails().positionType
+    let rate: bigint
+    if (positionType === PositionType.Supply) {
+      ;[rate, ,] = await lensContract.getAverageSupplyRatePerBlock(
+        protocolTokenAddress,
+        {
+          blockTag: blockNumber,
+        },
+      )
+    } else {
+      ;[rate, ,] = await lensContract.getAverageBorrowRatePerBlock(
+        protocolTokenAddress,
+        {
+          blockTag: blockNumber,
+        },
+      )
+    }
+    rate = rate * BigInt(AVERAGE_BLOCKS_PER_DAY[this.chainId]) * 365n
+    return Number(rate) / WAD
+  }
+
+  async getApr({
+    protocolTokenAddress,
+    blockNumber,
+  }: GetAprInput): Promise<ProtocolTokenApr> {
+    const apr = await this._getProtocolTokenApr({
+      protocolTokenAddress,
+      blockNumber,
+    })
+    return {
+      ...(await this._fetchProtocolTokenMetadata(protocolTokenAddress)),
+      aprDecimal: Number(apr) * 100,
+    }
+  }
+
+  async getApy({
+    protocolTokenAddress,
+    blockNumber,
+  }: GetApyInput): Promise<ProtocolTokenApy> {
+    const apr = await this._getProtocolTokenApr({
+      protocolTokenAddress,
+      blockNumber,
+    })
+
+    const apy = aprToApy(apr, SECONDS_PER_YEAR)
```

Comment:
```suggestion
    const apy = aprToApy(apr * 86400, 365)
```
This is the one used in the mc2 dapp (and compound app)